From 730be61e6645e72bcf70a8088a177cc43dd009d4 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@centricular.com>
Date: Wed, 29 Oct 2025 13:58:23 -0300
Subject: [PATCH 2/4] cmake: Don't modify CMAKE_MODULE_PATH before running
 project()

I was running a patch (amyspark/cerbero@dd3bff15) and this shows that the platform include (CMAKE_SYSTEM_INFO_FILE)
is taking effect before the project has even been set up.
See https://github.com/v-novaltd/LCEVCdec/blob/8d67e36f61df8d408c2058fc2206dbc2cda8d4b7/cmake/modules/CMakeProject.cmake#L48.
My understanding is that somehow the `list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules")`
is making the CMake internal code setting up the target platform/compiler prefer the project's platform code
instead of CMake's. Which, as remarked on Matrix, doesn't seem to happen on local machines.
---
 CMakeLists.txt                   | 5 +++--
 cmake/modules/CMakeProject.cmake | 6 +++---
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c69a6cc..0b174ae 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,8 +21,7 @@ option(VN_STRIP_RELEASE "Strip any debug information from release binaries" ON)
 option(VN_SDK_SIMD "Enable SIMD for current architecture (SSE + AVX2 or NEON)" ON)
 option(VN_SDK_LTO "Enable link time optimisation in the linker" OFF)
 
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules")
-include("CMakeSetup")
+include("${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/CMakeSetup.cmake")
 
 project(
     LCEVCdec_SDK
@@ -30,6 +29,8 @@ project(
     VERSION 4.0.3
     LANGUAGES CXX C)
 
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules")
+
 # Build configuration
 option(VN_SDK_EXECUTABLES "Include samples and test harness executables in SDK installations" OFF)
 option(VN_SDK_SAMPLE_SOURCE "Include sample sources in SDK installations" ON)
diff --git a/cmake/modules/CMakeProject.cmake b/cmake/modules/CMakeProject.cmake
index ede3b78..6db12c2 100644
--- a/cmake/modules/CMakeProject.cmake
+++ b/cmake/modules/CMakeProject.cmake
@@ -115,6 +115,6 @@ if (VN_SDK_LTO)
 endif ()
 
 #
-include("Arch/${TARGET_ARCH}" OPTIONAL)
-include("Platform/${TARGET_PLATFORM}")
-include("Compiler/${TARGET_COMPILER}")
+include("${CMAKE_CURRENT_LIST_DIR}/Arch/${TARGET_ARCH}.cmake" OPTIONAL)
+include("${CMAKE_CURRENT_LIST_DIR}/Platform/${TARGET_PLATFORM}.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/Compiler/${TARGET_COMPILER}.cmake")
-- 
2.51.1

